linera-execution/src/system.rs

```diff
@@ -486,17 +477,13 @@ where
                     ExecutionError::AlreadySubscribedToChannel(context.chain_id, channel)
                 );
                 self.subscriptions.insert(&subscription)?;
-                let message = RawOutgoingMessage {
-                    destination: Destination::Recipient(chain_id),
-                    authenticated: false,
-                    grant: Amount::ZERO,
-                    kind: MessageKind::Protected,
-                    message: SystemMessage::Subscribe {
-                        id: context.chain_id,
-                        subscription,
-                    },
+                let message = SystemMessage::Subscribe {
+                    id: context.chain_id,
+                    subscription,
                 };
-                outcome.messages.push(message);
+                txn_tracker.add_outgoing_message(
+                    OutgoingMessage::new(chain_id, message).with_kind(MessageKind::Protected),
```

Comment:
Are some messages `Protected` by design? Or should they be? I can imagine `Subscribe` could be one.